### PR TITLE
Multistore - International > Localization > Localization - Add checkboxes #19370

### LINKED ISSUE #19322: Multistore - International > Localization > Localization - Add specific settings dropdown
As a merchant, I want to be able to see if a parameter has a different value for another store when I am in "all shops" context or in a group context, so I can know to which stores the change I am making will be applied

[Specs specific settings dropdown](https://github.com/PrestaShop/prestashop-specs/blob/9783996765a46d3d7197e4f7cd7098ffb7e37ff2/content/1.7/back-office/multistoregeneralspecs.md#specific-settings-drop-down-only-from-178)

### LINKED ISSUE #19370: Multistore - International > Localization > Localization - Add checkboxes
As a merchant, I want to be able to select one or many fields / toggles when I am in a specific shop context or in a group context, so I can apply a different configuration for the selected store or group and see which parameters have different value(s) for the selected store or group.

[Specs checkboxes](https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/multistoregeneralspecs.md#checkboxes)

As specified in the [multistore specials cases specs](https://github.com/PrestaShop/prestashop-specs/blob/master/broader-topics/multistorespecialsspecs.md#list-of-pages--fields-for-specific-case-4-if-a-group-or-a-specific-shop-is-selected-some-fields-are-disabled-and-dont-have-checkboxes-allowing-to-enable-them), if a group or a specific shop is selected, some fields are disabled and don’t have checkboxes allowing to enable them on this page.

Add a message when the user hovers these fields: "You can modify this parameter in all shops context only. Switch context to enable it."

List of these fields:
- Time zone
- Language identifier
- Country identifier

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | As a merchant, I want to be able to select one or many fields / toggles when I am in a specific shop context or in a group context, so I can apply a different configuration for the selected store or group and see which parameters have different value(s) for the selected store or group.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/19370 Fixes https://github.com/PrestaShop/PrestaShop/issues/19322
| How to test?      | See issue 
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

:hammer: **Changed configuration forms**

- [x] AdvancedConfiguration Form
- [x] LocalizationConfiguration Form
- [x] LocalUnitsConfiguration Form

:information_source:  **Informations BC Break**

- Constructor `AdvancedConfiguration` changes
- Constructor `LocalizationConfiguration` changes
- Constructor `LocalUnitsConfiguration` changes
- Remove `AdminModuleDataProvider` from `LocalizationConfiguration`

:camera: **Screenshot**

Capture d’écran de 2022-01-30 18-50-39
Capture d’écran de 2022-01-30 18-50-59

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27533)

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Fail with reviewvable.io

File: tests/Unit/Adapter/Localization/LocalizationConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        $this->mockShopConfiguration = $this->createShopContextMock();
+        $this->mockMultistoreFeature = $this->createMultistoreFeatureMock();
+        $this->mockLanguageActivator = $this->createLanguageActivatorMock();
+        $this->mockCurrencyManager = $this->createCurrencyManagerMock();
+    }
+
+    /**
+     * @return LanguageActivatorInterface
+     */
+    protected function createLanguageActivatorMock()
```
Comment (Progi1984) [inline tests/Unit/Adapter/Localization/LocalizationConfigurationTest.php]:
```suggestion
    protected function createLanguageActivatorMock(): LanguageActivatorInterface
```

File: tests/Unit/Adapter/Localization/LocalizationConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    {
+        return $this->getMockBuilder(LanguageActivatorInterface::class)
+            ->getMock();
+    }
+
+    /**
+     * @return CurrencyManager
+     */
+    protected function createCurrencyManagerMock()
```
Comment (Progi1984) [inline tests/Unit/Adapter/Localization/LocalizationConfigurationTest.php]:
```suggestion
    protected function createCurrencyManagerMock(): CurrencyManager
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback but it's approved

**Comment by ghost** [comment]
Done @Progi1984

**Review by ghost — COMMENTED** [review]
Edit : reviewvable.io comment

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]
nice @okom3pom 👍

**Review by marwachelly — DISMISSED** [review]
Hello,

Thank you for the PR. Could you please check the specification in the https://github.com/PrestaShop/PrestaShop/issues/19370 for the fields:

- Time zone
- Language identifier
- Country identifier

According to the spec those field should be disabled and don’t have checkboxes allowing to enable them on this page and a message when the user hovers these fields: "You can modify this parameter in all shops context only. Switch context to enable it."

https://github.com/orgs/PrestaShop/teams/product-team Could you please specify if the section Import localisation pack should have checkboxes or it can be modified only in all shop contex?

The actual behavior in the PR is as follow:
https://watch.screencastify.com/v/uRWraC5elpKvd12KZKSx

Thanks,

**Comment by ghost** [comment]
Hello @marwachelly 

It should be possible for the time zone to be different for each shop.

I have one shop in London and the other in Paris, it is not the same time zone.

It's the same for: Language identifier and Country identifier

Many shops use ipfailover to have an ip in the country where they sell

@MatShir what do you think?

At the beginning of the video the import part is not multishop because we have to activate or deactivate the languages in the shps, these options can't be multishop

**Comment by MatShir** [comment]
@okom3pom that makes a lot of sense. A cart rule depending on the time zone of the shop can differ a lot. Very good point. I'm going to update the specs but to make it work, we need to check that the features of the timezone work properly.

@marwachelly into the tests, we need to add:
- Create a Cart Rule in the different shops and check if it adapts for the different timezone and if the date of the creation of order matches the timezone set in the different shops
- Check if the language adapts to different shops
- Check if the country adapts to different shops

**Review by marwachelly — COMMENTED** [review]
Hello @okom3pom,

Could you please check the conflicting files?

Thanks,

**Comment by matks** [comment]
Hi @okom3pom 

Since we had no news from you since March, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Comment by ghost** [comment]
I am on it

**Comment by Progi1984** [comment]
@okom3pom asked me to reopen it.

**Review by matthieu-rolland — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

**Comment by kpodemski** [comment]
Hey @okom3pom 

This PR was in QA already, did you close it by mistake?

**Review by khouloudbelguith — COMMENTED** [review]
Hi @okom3pom,

I have an issue following these steps:

1. Fresh install of this PR with English language & France country
2. Add new currency for example `DZD`
3. Enable multistore > Create a new shop & configure it
4. Go to BO > International > Localization page
5. Select any shop, for example, `shop2`
6. Select "Default currency" & click on Save without changing it
7. See error > the checkbox "Default currency"  is still disabled
8. Select "All shops" 
9. See error > The dropdown list is not displayed for the field "Default currency" 
10. Change the "Default currency" to DZD
11. Save & check the shop2, the currency is not changed

I attached a screen record

https://user-images.githubusercontent.com/16067358/201369880-66adcbc3-4eb1-4b68-a36f-359100c2bb82.mp4

Thanks!

**Pull request closed without merging**